packages/data/src/components/use-select/test/index.js

```diff
@@ -565,5 +568,50 @@ describe( 'useSelect', () => {
 				childCount: 0,
 			} );
 		} );
+
+		it( 'handles non-existing stores', () => {
+			registry.registerStore( 'store-1', counterStore );
+
+			let renderer;
+
+			const TestComponent = jest.fn( () => {
+				const state = useSelect(
+					( select ) => ( {
+						count1: select( 'store-1' ).getCounter(),
+						blank: select( 'non-existing-store' )?.getCounter(),
```

Comment:
Can we show a `console.warn` when selecting a non-existing store? Similar to as we do when [registering blocks with invalid category](https://github.com/WordPress/gutenberg/blob/2239c5d7e3bd1ef9e7b3362220c6dff43ef8e1f8/packages%2Fblocks%2Fsrc%2Fapi%2Fregistration.js#L244-L258). This way we'd also immediately see that a store is not available (whatever the reason), which could help to track issues that it might cause.

Reply:
This has been brought up internally, c.c. @youknowriad.

The counter point is that sometimes it's intended, as shown in the example in the PR description. I'm not sure if it's an anti-pattern or not, but this pattern does exist. I think it needs further discussion and more usage examples.

Reply:
Ah, gotcha. Maybe we could consider adding sth like `isStoreRegistered` to `@wordpress/data` then. This does sound like a topic for another issue though 👍